### PR TITLE
[PLATFORM-155] - Docs

### PR DESCRIPTION
**Pull request by fonty1**
First bite at pushing real content inside the `docs/` using [MDX](https://github.com/mdx-js/mdx). Docs lives in its own folder, and has its own routes, pages, components, layout and navigation semi-independent of the main app. 

**Docs Styling**
General styling is inside `docs/components/DocsLayout/`

**Main Navigation**
Located in `docs/components/DocsLayout/Navigation` it iterates through the navLinks object to render the primary navigation. 

**InPage Navigation**
The nested sub navigation items are defined per page and passed as props to the Navigation component. The subNav object contains the Headings text and IDs which correlate to the ScrollableAnchor sections inside the `content.mdx` files. Since the IDs need to be synced manually there may be a better way to generate these but this works well for now. 

**Mobile Navigation** 
Same Navigation component used here with a format prop toggle for mobile as it needs a slightly different render.
https://share.goabstract.com/e6f1db79-5cd6-4fc2-883c-f7d679dca42f

Please review the implementation of:
- Desktop, Mobile, in-page navigation
- General organisation of the Docs/Content

I have created the following Jira tasks for the remaining work, best done after this has been merged:

- Code highlighting snippet component - https://streamr.atlassian.net/browse/PLATFORM-310
- Warning component - https://streamr.atlassian.net/browse/PLATFORM-158 
- Page Turner component - https://streamr.atlassian.net/browse/PLATFORM-157
- Generate Tutorials Page Content - https://streamr.atlassian.net/browse/PLATFORM-311
- Generate Getting Started Page Content - https://streamr.atlassian.net/browse/PLATFORM-312
- Generate Visual Editor Page Content - https://streamr.atlassian.net/browse/PLATFORM-313
- Generate Streamr APIs Page Content - https://streamr.atlassian.net/browse/PLATFORM-314 

Content for the Introduction page and Marketplace pages is largely done (as a first pass at least) but please add anything I've missed in Jira.

### CODE REVIEW

File: app/src/docs/components/ApiPage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// @flow
+
+import React from 'react'
+
+import DocsLayout from '../DocsLayout'
+// $FlowFixMe
+import ApiPageContent from './Content.mdx'
```
Comment (juhah) [inline app/src/docs/components/ApiPage/index.jsx]: Could these import errors be fixed by editing the `.flowconfig`?

File: app/src/docs/components/ApiPage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+const subNav = {
+    clientLibraries: 'Client Libraries',
+    authentication: 'Authentication',
```
Comment (juhah) [inline app/src/docs/components/ApiPage/index.jsx]: Maybe these could be read automatically by traversing `ScrollableAnchor` elements in the child page.

File: app/src/docs/components/ButtonBookmark/index.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import * as React from 'react'
+
+import styles from './buttonBookmark.pcss'
+
+type Props = {
+    children: React.Node,
+}
+
+const ButtonBookmark = ({ children }: Props) => (
```
Comment (juhah) [inline app/src/docs/components/ButtonBookmark/index.jsx]: I think `Panel` or something like that might be a more appropriate name?

File: app/src/docs/components/DocsLayout/Navigation/index.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    location: Location,
+}
+
+type State = {
+    compressed: boolean,
+}
+
+class Navigation extends React.Component<Props, State> {
+    state = {
+        compressed: true,
```
Comment (juhah) [inline app/src/docs/components/DocsLayout/Navigation/index.jsx]: Compressed = collapsed?

File: app/src/docs/components/DocsLayout/Navigation/index.jsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                ) : null
+        ))
+    }
+
+    render() {
+        const { className, format } = this.props
+
+        return (
+            <div // eslint-disable-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+                className={cx(className, styles.navigationContainer, this.state.compressed ? styles.compressed : '')}
```
Comment (juhah) [inline app/src/docs/components/DocsLayout/Navigation/index.jsx]: Can be formatted as:
```
className={cx(className, styles.navigationContainer, {
    [styles.compressed]: this.state.compressed,
})}
```

File: app/src/docs/components/DocsLayout/Navigation/index.jsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                onClick={this.toggleExpand}
+            >
+                <ul className={styles.navList}>
+                    {format === 'mobile' ? this.parseCurrentPage() : false}
```
Comment (juhah) [inline app/src/docs/components/DocsLayout/Navigation/index.jsx]: `{format === 'mobile' && this.parseCurrentPage()}`?

File: app/src/docs/components/ApiPage/index.jsx (hunk 2 above)
Comment (fonty1) [reply app/src/docs/components/ApiPage/index.jsx]: I'm not sure how to do this in a straight forward way so lets leave it for now and come back to it in a future ticket.

**Pull request merged**